### PR TITLE
Fixes for mobile datepicker

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Fix closing the datepicker after "go-next" on duration field on mobile devices due to a wrong button type on cancel button
- [x] Treat mobile datepicker modal as other modals, being more or less fullscreen with a black background (and close button for the scope of this PR)
- [x] Add CDK focus trapping on the datepicker for good measure
- [x] The start date, finish date and duration fields open the numeric keyboard, not alpha-numeric one.

https://community.openproject.org/projects/openproject/work_packages/44144/activity?query_id=3510

### CODE REVIEW

**Comment by lindenthal** [comment]
Hi @oliverguenther and @psatyal,

I just had a quick preview on iOS FF:
- Is it possible to use the entire width?
- Do we need two close actions?

Can you please compare with Google Flights on mobile?

Best
Niels


EAF7B73C-2B46-4601-B8ED-B5C46C303A75

**Comment by oliverguenther** [comment]
@lindenthal, width of the datepicker is fixed by flatpickr. It doesn't support autoscaling, and implementing this ourselves is unfeasible.

The `X` close is shown for all mobile modals. We could hide the cancel button for mobile?

**Comment by oliverguenther** [comment]
I extracted two bugs from this PR:

- The two cancel/close button issue: https://community.openproject.org/wp/44398
- Question regarding alignment: https://community.openproject.org/wp/44399

File: frontend/src/app/spot/components/text-field/text-field.component.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export class SpotTextFieldComponent implements ControlValueAccessor {
 
   @Input() public value = '';
 
+  @Input() public pattern:string|undefined;
+
+  @Input() public inputmode:string = 'text';
```
Comment (b12f) [inline frontend/src/app/spot/components/text-field/text-field.component.ts]: Could you add doc comments to both of these? If you merge `release/12.3`, you should see the others that are now there.

**Review by b12f — APPROVED** [review]
LGTM

File: frontend/src/app/spot/components/text-field/text-field.component.ts (hunk 1 above)
Comment (oliverguenther) [reply frontend/src/app/spot/components/text-field/text-field.component.ts]: Done 👍

**Pull request merged**